internal/indexnode/indexnode.go

```diff
@@ -220,6 +223,18 @@ func (i *IndexNode) Start() error {
 
 // Stop closes the server.
 func (i *IndexNode) Stop() error {
+	if i.stopFlag.Load() {
```

Comment:
Should set sate code to StateCode_Stopping state before starting the stop method.